### PR TITLE
Update pinned dependencies

### PR DESCRIPTION
**Pull request by khallock**
The pinned dependencies have been updated to address issues with an incompatible version of the icu package.

### CODE REVIEW

**Comment by khallock** [comment]
Is it possible to keep $SRC_DIR around for testing? The test this recipe uses exists in the source code, but isn't installed.
```
+ /bin/bash -x -e /feedstock_root/build_artefacts/recipe_root_1486502760902/test_tmp/run_test.sh
+ cd /feedstock_root/build_artefacts/recipe_root_1486502760902/test_tmp/src/examples
/feedstock_root/build_artefacts/recipe_root_1486502760902/test_tmp/run_test.sh: line 3: cd: /feedstock_root/build_artefacts/recipe_root_1486502760902/test_tmp/src/examples: No such file or directory
```

**Comment by ocefpaf** [comment]
We need to use https://github.com/conda-forge/compliance-checker-feedstock/blob/master/recipe/meta.yaml#L36 now. Latest `conda-build` no longer keeps the source dir by default.

**Comment by khallock** [comment]
Thanks for the suggestion, @ocefpaf! The recipe seems to be working now (on my machine at least).

**Comment by khallock** [comment]
The Mac build seems to be failing during testing:
```
ImportError: dlopen(/Users/travis/miniconda3/conda-bld/recipe_1486628260971/_t_env/lib/python2.7/site-packages/PyNGL/_hlu.so, 2): Symbol not found: _cairo_xlib_surface_create
  Referenced from: /Users/travis/miniconda3/conda-bld/recipe_1486628260971/_t_env/lib/python2.7/site-packages/PyNGL/_hlu.so
```
I'm not sure where "_cairo_xlib_surface_create" is supposed to be coming from, but this was not an issue on the Linux build.

**Comment by david-ian-brown** [comment]
HI Kevin,
I think "_cairo_xlib_surface_create" is in the cairo library. It might be
because XQuartz changed its location from /usr/X11/ to /opt/X11/. The cairo
libraries on mac are located with XQuartz. I don't know if this has
anything to do with the conda-forge set up though.
 -dave

**Comment by khallock** [comment]
Hi Dave,
Unfortunately, the new Mac build image on Travis CI no longer has XQuartz installed by default. conda-forge also recently added several xorg-* packages, which is one of the things I'm testing in this pull request.

@ocefpaf, is it possible to install XQuartz in the build environment? Or should we try our best to get the xorg-* packages to work?

**Comment by ocefpaf** [comment]
> @ocefpaf, is it possible to install XQuartz in the build environment? Or should we try our best to get the xorg-* packages to work?

In theory using all `conda-forge` packages would give is more control instead of relying on `Travis-CI` image. But I am not an `OS X` expert and I would be in favor of the easiest solution.

Pinging @pkgw, @ccordoba12, and @mingwandroid for their opinion. (It would be nice if we could standardize this and use whatever we decide on qt/pyqt too.)

**Comment by ccordoba12** [comment]
> It would be nice if we could standardize this and use whatever we decide on qt/pyqt too

Neither Qt 4 nor 5 depend on Xorg/XQuartz in macOS

**Comment by ocefpaf** [comment]
> Neither Qt 4 nor 5 depend on Xorg/XQuartz in macOS

The xorg packages are not only for `OS X`, we can use them in the Linux packages too and, right now, we user `yum_requirements.txt` for `Qt4` on Linux to get the X11 stuff.

TL;DR I meant that we should standardize how we use the xorg vc  `yum_requirements.txt`.

**Comment by khallock** [comment]
I tested this on my machine by (locally) building cairo with re-enabled X11 support on Mac, and then using that to build PyNGL. I was then able to run an example PyNGL script that opens an X11 display using XQuartz with no issues.

Once cairo-feedstock is updated, we should hopefully be able to test and merge this PR fairly quickly.

**Comment by pkgw** [comment]
Xref [cairo-feedstock #22](https://github.com/conda-forge/cairo-feedstock/issues/22), discussing the addition of X.org deps to `cairo`.

**Comment by ocefpaf** [comment]
This is outdated now with the new pinning scheme already in place for this feedstock.
